tasks/install.yml

```diff
@@ -85,26 +85,39 @@
         update_cache: true
       check_mode: false
 
-- name: Ensure gitlab-rails file exist
+- name: Check if VERSION file exists
   ansible.builtin.stat:
-    path: "/usr/bin/gitlab-rails"
-  register: file
+    path: "/var/opt/gitlab/gitlab-rails/VERSION"
+  register: gitlab_version_file
 
-- name: Wait until all previous background migrations are processed.
-  become: yes
-  ansible.builtin.command:
-    cmd: gitlab-rails runner -e production 'puts Gitlab::BackgroundMigration.remaining'
-  register: remaining
-  until: remaining.stdout == "0"
-  retries: 10
-  delay: 10
-  check_mode: no
-  changed_when: remaining.rc != 0
+- name: Wait for background migrations on primary node
   when:
-    - file.stat.exists
-    - file.stat.executable
-    - gitlab_is_primary is defined
     - gitlab_is_primary
+    - gitlab_version_file.stat.exists
+    - gitlab_rails_binary.stat.exists
+    - gitlab_rails_binary.stat.executable
```

Comment:
To better understand: Is this placed for safety?

Is there a situation in which we have the version file and not gitlab rails binary?

Reply:
> Is there a situation in which we have the version file and not gitlab rails binary?

I can't say for sure. Since it was there before and it does not hurt at the moment, I kept it.